### PR TITLE
Sm03lebr00t/feature/suggest side

### PR DESCRIPTION
**Pull request by ArcticSpaceFox**
## What does this do?

this adds the `suggest side` redemption, which allows viewers to suggest a side project. The suggestion will be added to a specified file.

## What can be improved?

- [x] retrieve file path from somewhere else
- [ ] add propper logging

### CODE REVIEW

**Comment by Sm03leBr00t** [comment]
✅ PR no conflicts, code looks ok, merging...

File: src/channel_points_redemption.rs [hunk 1]
```diff
@@ -28,6 +28,37 @@ pub fn channel_points_redemption(res_msg: &Res) {
...
+    // [SuggestSide] - redemtion, will add a side, if redeemed, to a specified file
+    // if file does not exist it will create file, otherwise it will truncate to it
+    if redemption_title.contains("Suggest Side") {
+        // Check if a file path has been supplied, if not use a default path
+        let _settingsMap = settings.try_into::<HashMap<String, String>>().unwrap();
+        let filepath = match _settingsMap.get("side_suggestions_file") {
+            Some(value) => value,
+            None => {
+                // If there is no specified path, we will create a temporary directory and push a default file name in there
+                let mut tempdir = temp_dir();
```
Comment (MathiasKandelborg) [inline src/channel_points_redemption.rs]: temp_dir what 🥰

**Review by MathiasKandelborg — CHANGES_REQUESTED** [review]
🥰

File: src/twitch.rs [hunk 2]
```diff
@@ -66,12 +67,13 @@ pub fn setup_twitch_chat_ws() -> TwidshTshadBott {
...
 
 impl TwidshTshadBott {
-    pub fn main(&mut self, commands: &Config) {
+    pub fn main(&mut self, commands: &Config, settings: &Config) {
         if let Err(Disconnected) = self.read_message(commands) {
             self.back_off()
         } else {
            // println!("Chat read msg successful");
         }
+        self.settings = settings;
```
Comment (MathiasKandelborg) [inline src/twitch.rs]: We wanna have it in the else :D

File: src/channel_points_redemption.rs (hunk 1 above)
Comment (ArcticSpaceFox) [reply src/channel_points_redemption.rs]: will create a temporary PathBuf

File: src/twitch.rs (hunk 2 above)
Comment (ArcticSpaceFox) [reply src/twitch.rs]: maybe maybe not I can change that

**Review by MathiasKandelborg — APPROVED** [review]

**Pull request merged**